src/transformers/modeling_tf_t5.py

```diff
@@ -791,3 +791,22 @@ def call(self, decoder_input_ids, **kwargs):
         decoder_outputs = (lm_logits,) + decoder_outputs[1:]
 
         return decoder_outputs + encoder_outputs
+
+    def prepare_inputs_for_generation(self, input_ids, past, attention_mask, **kwargs):
+        assert past is not None, "past has to be defined for encoder_outputs"
+
+        # first step
+        if type(past) is tuple:
+            encoder_hidden_states = past[0]
+        else:
+            encoder_hidden_states = past
+
+        return {
+            "inputs": input_ids,
+            "encoder_hidden_states": encoder_hidden_states,
+            "encoder_attention_mask": attention_mask,
+        }
+
+    def _reorder_cache(self, past, beam_idx):
```

Comment:
I would be fine having this be the default on the parent

Reply:
Even though this looks very basic, this is only the default case for encoder-decoder models, the current default is the default for all "decoder" only models - so I'd prefer to leave it as it is.